lib/kafka/consumer.rb

```diff
@@ -280,6 +280,21 @@ def each_batch(min_bytes: 1, max_wait_time: 5, automatically_mark_as_processed:
       end
     end
 
+    # @param min_bytes [Integer] the minimum number of bytes to read before
+    #   returning messages from the server; if `max_wait_time` is reached, this
+    #   is ignored.
+    # @param max_wait_time [Integer, Float] the maximum duration of time to wait before
+    #   returning messages from the server, in seconds.
+    # @param topics_offsets [Hash] the offsets to fetch from. By default it will
+    #   fetch from where the offset manager defaults too.
+    # @return batch [Kafka::FetchedBatch] a message batch fetched from Kafka.
+    def poll(min_bytes: 1, max_wait_time: 5, offsets: {})
```

Comment:
I've been thinking about adding seeking to the normal consumer API as well, e.g.

```ruby
offsets = MyDB.find_offsets_for("my-topic")

offsets.each do |partition, offset|
  # Seeks the specified topic/partition to `offset`
  consumer.seek("my-topic", partition, offset)
end

consumer.each_message {|message| ... }
```

The main issue is whether _all_ consumers in a group need to do this or if we should propagate the specified offsets somehow – maybe using the `Sync` Kafka API.